src/shared/components/action-buttons/ShareDropdown.tsx

```diff
@@ -0,0 +1,138 @@
+import { Dispatch, MouseEvent, SetStateAction } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import { DropdownButton, Button, CopyIcon } from 'franklin-sites';
+import { createPath } from 'history';
+
+import useNS from '../../hooks/useNS';
+import useLocalStorage from '../../hooks/useLocalStorage';
+
+import { addMessage } from '../../../messages/state/messagesActions';
+
+import { Namespace } from '../../types/namespaces';
+import {
+  MessageFormat,
+  MessageLevel,
+} from '../../../messages/types/messagesTypes';
+import { Column, nsToDefaultColumns } from '../../config/columns';
+import { ViewMode } from '../results/ResultsData';
+
+const isCopySupported =
+  'clipboard' in navigator && 'writeText' in navigator.clipboard;
+
+// TODO: expose way to close dropdown (in Franklin)
+const clickOnDropdown = (element: HTMLElement) => {
+  (
+    element.closest('.dropdown-container')?.firstElementChild as
+      | HTMLElement
+      | null
+      | undefined
+  )?.click();
+};
+
+const CopyLinkWebsite = () => {
+  const dispatch = useDispatch();
+  const ns = useNS() || Namespace.uniprotkb;
+  const [userColumns] = useLocalStorage<Column[]>(
+    `table columns for ${ns}` as const,
+    nsToDefaultColumns(ns)
+  );
+  const [viewMode] = useLocalStorage<ViewMode>('view-mode', ViewMode.CARD);
```

Comment:
In four locations in the codebase we specify `ViewMode.CARD` as the default view mode. I am wondering if it would be safer to create a `defaultViewMode = ViewMode.CARD` in one location and use this instead.

Reply:
OK, with that.
I would also take the opportunity to change to an enum here too. So it would be replacing the current 0 and 1 by 'card' and 'table' (in localStorage and in the shared URL). Do you think it would be acceptable to do?